### PR TITLE
feat(web): lift the routes to app

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
As an enabler for #7309 we need to have context of the routes in the getState API call which is triggered at the top of the tree. 

## Description
This PR lifts our routes to app.tsx for we have context of the params across the app.

### CODE REVIEW

File: web/src/features/panels/LeftPanel.tsx [hunk 1]
```diff
@@ -117,37 +61,12 @@ function OuterPanel({ children }: { children: React.ReactNode }) {
       } ${isOpen ? '' : '-translate-x-full'}`}
     >
       {isMobile && <MobileHeader />}
-      <section className="h-full w-full">{children}</section>
+      <section className="h-full w-full">
+        <Suspense fallback={<LoadingSpinner />}>
+          {isZonePath ? <ZoneDetails /> : <RankingPanel />}
+        </Suspense>
```
Comment (VIKTORVAV99) [inline web/src/features/panels/LeftPanel.tsx:67]: While this works would it not be better to use a outlet and pass the component down from the router?

https://reactrouter.com/en/main/components/outlet#outlet

File: web/src/App.tsx [hunk 2]
```diff
@@ -113,7 +157,20 @@ export default function App(): ReactElement {
...
+                      path="/zone/:zoneId/:urlTimeAverage?/:urlDatetime?"
+                      element={
+                        <ValidZoneIdGuardWrapper>
+                          <LeftPanel />
+                        </ValidZoneIdGuardWrapper>
+                      }
+                    />
+                    <Route path="*" element={<LeftPanel />} />
+                  </Routes>
+                </Suspense>
```
Comment (VIKTORVAV99) [inline web/src/App.tsx:173]: Then this would be more declarative and we could avoid using the left panel twice.

File: web/src/App.tsx [hunk 3]
```diff
@@ -31,20 +33,57 @@ const TimeControllerWrapper = lazy(() => import('features/time/TimeControllerWra
...
+  if (zoneId !== upperCaseZoneId) {
+    return <Navigate to={`/zone/${upperCaseZoneId}?${searchParameters}`} replace />;
+  }
+
+  // Handle legacy Australia zone names
+  if (upperCaseZoneId.startsWith('AUS')) {
+    return (
+      <Navigate to={`/zone/${zoneId.replace('AUS', 'AU')}?${searchParameters}`} replace />
+    );
+  }
```
Comment (VIKTORVAV99) [inline web/src/App.tsx:74]: Just going to ask this while we are at it:

Should we move this to a redirect rule in Cloudflare instead?

**Review by VIKTORVAV99 — COMMENTED** [review]
Works as I expect but I have some questions about the logic behind it.

Did you happen to look into https://reactrouter.com/en/main/routers/picking-a-router#using-v64-data-apis? I and Cadence want to switch to using data routing so we can use the fancy data APIs.

**Comment by tonypls** [comment]
> Works as I expect but I have some questions about the logic behind it.
> 
> Did you happen to look into https://reactrouter.com/en/main/routers/picking-a-router#using-v64-data-apis? I and Cadence want to switch to using data routing so we can use the fancy data APIs.

Good point! and thanks for the feedback, I'll close this one and let's revisit with a better approach.

**Pull request closed without merging**